File: app/src/main/java/com/nbennettsoftware/android/npad/SettingsActivity.java
package com.nbennettsoftware.android.npad;


import android.os.Bundle;

import com.nbennettsoftware.android.npad.widgets.WallpaperView;

public class SettingsActivity extends NpadActivity
        implements SettingsFragment.OnWallpaperChangedListener,
                    SettingsFragment.OnShadeChangedListener,
                    SettingsFragment.OnScalingChangedListener{

    private WallpaperView wallpaperView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);



        SettingsFragment settingsFragment = new SettingsFragment();
        settingsFragment.setOnWallpaperChangedListener(this);
        settingsFragment.setOnShadeChangedListener(this);
        settingsFragment.setOnScalingChangedListener(this);

        getFragmentManager().beginTransaction()
                .replace(R.id.settings_fragment_placeholder, settingsFragment)
                .commit();


        wallpaperView = findViewById(R.id.settings_wallpaperImageView);

        refreshUi();
    }

    @Override
    public void OnWallpaperChanged() {
        refreshUi();
    }

    @Override
    public void OnDimmerChanged(String dimmerIntensity) {
        wallpaperView.applyWallpaperDimmer(dimmerIntensity);
    }

    @Override
    public void OnScalingChanged(String scaling) {
        wallpaperView.applyScaling(scaling);
    }

    void refreshUi(){
        wallpaperView.applyWallpaperFromPreferences();
        wallpaperView.applyWallpaperDimmerFromPreferences();
        wallpaperView.applyScalingFromPreferences();
    }

    @Override
    protected void onResume() {
        super.onResume();
        refreshUi();
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
    }

}

File: app/src/main/java/com/nbennettsoftware/android/npad/SettingsActivity.kt
package com.nbennettsoftware.android.npad


import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity

import com.nbennettsoftware.android.npad.widgets.WallpaperView

class SettingsActivity : AppCompatActivity(), SettingsFragment.OnWallpaperChangedListener, SettingsFragment.OnShadeChangedListener, SettingsFragment.OnScalingChangedListener {

    private var wallpaperView: WallpaperView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)


        val settingsFragment = SettingsFragment()
        settingsFragment.setOnWallpaperChangedListener(this)
        settingsFragment.setOnShadeChangedListener(this)
        settingsFragment.setOnScalingChangedListener(this)

        supportFragmentManager.beginTransaction()
                .replace(R.id.settings_fragment_placeholder, settingsFragment as Fragment)
                .commit()


        wallpaperView = findViewById(R.id.settings_wallpaperImageView)

        wallpaperView?.applyWallpaperFromPreferences()
        wallpaperView?.applyWallpaperDimmerFromPreferences()
        wallpaperView?.applyScalingFromPreferences()

    }

    override fun OnWallpaperChanged() {
        wallpaperView?.applyWallpaperFromPreferences()
    }

    override fun OnDimmerChanged(intensity: String) {
        wallpaperView?.applyWallpaperDimmer(intensity)
    }

    override fun OnScalingChanged(scaling: String) {
        wallpaperView?.applyScaling(scaling)
    }

    override fun onBackPressed() {
        super.onBackPressed()
        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
    }

}
